Console/Command/SynonymDeduplicateCommand.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Console\Command;
+
+use Algolia\AlgoliaSearch\Helper\AlgoliaHelper;
+use Algolia\AlgoliaSearch\Service\IndexNameFetcher;
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Magento\Framework\App\State;
+use Magento\Framework\Console\Cli;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SynonymDeduplicateCommand extends AbstractStoreCommand
+{
+    public function __construct(
+        protected AlgoliaHelper         $algoliaHelper,
+        protected IndexNameFetcher      $indexNameFetcher,
+        protected State                 $state,
+        protected StoreNameFetcher      $storeNameFetcher,
+        protected StoreManagerInterface $storeManager,
+        ?string                         $name = null
+    ) {
+        parent::__construct($state, $storeNameFetcher, $name);
+    }
+
+    protected function getCommandPrefix(): string
+    {
+        return parent::getCommandPrefix() . 'synonyms:';
+    }
+
+    protected function getCommandName(): string
+    {
+        return 'deduplicate';
+    }
+
+    protected function getCommandDescription(): string
+    {
+        return "Identify and remove duplicate synonyms in Algolia";
+    }
+
+    protected function getStoreArgumentDescription(): string
+    {
+        return 'ID(s) for store(s) containing synonyms to deduplicate in Algolia (optional), if not specified, synonyms for all stores will be deduplicated';
+    }
+
+    protected function getAdditionalDefinition(): array
+    {
+        return [];
+    }
+
+    /**
+     * @throws NoSuchEntityException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->input = $input;
+        $this->output = $output;
+        $this->setAreaCode();
+
+        if (!$this->confirmDedupeOperation()) {
+            return Cli::RETURN_SUCCESS;
+        }
+
+        $storeIds = $this->getStoreIds($input);
```

Comment:
Is `bin/magento algolia:synonyms:deduplicate 1 2` the standard way to expect an array of integers ? Maybe it would be useful to add an example in the `getStoreArgumentDescription` method ?

Reply:
C'est correct et je suis d'accord...
<img width="809" alt="image" src="https://github.com/user-attachments/assets/b9fe23ff-3916-45aa-b3bb-d8fa45f49a0f" />